generic_scripts/nlk.py

```diff
@@ -0,0 +1,305 @@
+
+    def calc_magnetic_field(self, r):
+        """Computes the magnetic field (in units of [T]) generated by the wire
+        in a set of points in space.
+
+        Parameters
+        ----------
+        r : numpy.array
+            Array of shape (2 x N) with the coordinates [[x1, x2, ... , xN],
+            [y1, y2, ... , yN]] where the field will be returned.
+
+        Returns
+        -------
+        numpy.array
+            Array with the same shape of r, containing the magnetic field
+            components at each point in r.
+
+        Raises
+        ------
+        ValueError
+            If input r hasn't shape of type (2 X N).
+        """
+        if r.shape[0] != 2:
+            raise ValueError(
+                "The input position vector r must have shape of type (2, N)")
+        r = r.reshape(2, -1)
+        mu0 = vacuum_permeability
+        r_w = self.position[:, None]  # Wire positions
+        rc = r - r_w  # Cursive r
+        theta = _np.arctan2(rc[1], rc[0])
+        theta_vec = _np.array([-_np.sin(theta), _np.cos(theta)])
+        rc_norm = _np.linalg.norm(rc, axis=0)[None, :]
+        mag_field = mu0 * self.current/(2*_np.pi*rc_norm)*theta_vec
+        return mag_field
+
+    @property
+    def position(self):
+        """."""
+        return self._pos.copy()
+
+    @position.setter
+    def position(self, position):
+        """."""
+        position = self._check_r_size(position)
+        self._pos = position
+
+    @property
+    def current(self):
+        """."""
+        return self._curr
+
+    @current.setter
+    def current(self, current):
+        """."""
+        self._curr = current
+
+    @staticmethod
+    def _check_r_size(r):
+        """."""
+        r = _np.asarray(r)
+        if r.size == 2:
+            return r.ravel()
+        else:
+            raise ValueError('r must be a numpy array of size 2.')
+
+
+class NLK:
+    """Simulates the magnetic field generated by the nonlinear kicker.
+    """
+    def __init__(self, positions=None, current=1850):
+        """Initialize a NLK object
+
+        Parameters
+        ----------
+        positions : numpy.array, optional
+            Array of shape (2, 8) with the wires positions [[x1, ... , xN],
+            [y1, ... , yN]], by default loads the nominal wire positions.
+
+        current : float, optional.
+            Current of the wires in Amperes, by default 1850 A.
+        """
+        if positions is not None:
+            wire_positions = positions
+        else:
+            wire_positions = _np.zeros([2, 8])
+            s1 = _np.array([1, -1, 1, -1])
+            s2 = _np.array([1, 1, -1, -1])
+            wire_positions[:, ::2] = _np.array([s1*7, s2*5.21])
+            wire_positions[:, 1::2] = _np.array([s1*10, s2*5.85])
+            wire_positions *= 1e-3
+
+        # Creating wires
+        currents = _np.zeros(8)
+        currents[::2] = -current
+        currents[1::2] = current
+
+        self._wires = [
```

Comment:
I think you don't need this line break here.

Reply:
Solved, thanks!